### PR TITLE
zstd: add concurrency option to codec

### LINKED ISSUE #123: Write columns in parallel
Our bottleneck in writing ~500MB zstd compressed files is that the write happens single threaded, often taking minutes to complete. It also takes ~15 GB of RAM during that processing, which is a weird ratio for cloud resources. Writing / compressing in parallel would significantly increase our throughput.

There's prior art in the rust implementation, which hopefully translates well to Go.
- https://github.com/apache/arrow-rs/pull/4871

### PR DESCRIPTION
**Pull request by derekperkins**
This seems pretty straightforward, and I copied the style for defining a level. The only question I had was whether it was worth creating separate encoder and decoder concurrency levels, but that seemed unnecessary. If someone really needs it, they can always use two separate Codec objects.

Fixes #123

### CODE REVIEW

File: compress/zstd/zstd.go [hunk 1]
```diff
@@ -31,11 +31,17 @@ const (
...
+
+	DefaultConcurrency = 1
 )
 
 type Codec struct {
 	Level Level
 
+	// Concurrency is the number of CPU cores to use for encoding and decoding.
+	// If Concurrency is 0, it will use DefaultConcurrency.
+	Concurrency int
```
Comment (fpetkovski) [inline compress/zstd/zstd.go:43]:
```suggestion
	Concurrency uint
```

Comment (derekperkins) [reply compress/zstd/zstd.go]: changed

**Review by fpetkovski — APPROVED** [review]
Nice

**Review by thorfour — APPROVED** [review]

**Comment by derekperkins** [comment]
Thanks!

**Pull request merged**